packages/public/common-ui-web/src/components/Slider/useSlider.ts

```diff
@@ -0,0 +1,170 @@
+}
+
+/**
+ * Props accepted by the userSlider hook.
+ */
+export interface useSliderProps {
+  /** Reference to the HTMLDivElement representing the slider. */
+  sliderRef: React.RefObject<HTMLDivElement>;
+
+  /** The current value of the slider. */
+  value: number;
+
+  /** The minimum value of the slider range. */
+  min: number;
+
+  /** The maximum value of the slider range. */
+  max: number;
+
+  /** The step value for the slider. */
+  step: number;
+
+  /** Indicates whether the slider is disabled or not. */
+  disabled: boolean;
+
+  /**
+   * Callback function invoked when the slider value changes.
+   * @param value - The new value of the slider.
+   */
+  onChange?: (value: number) => void;
+}
+
+/**
+ * Custom hook for creating a slider component with customizable behavior.
+ * @param options - The configuration options for the slider component.
+ * @returns An object containing the current thumb position, a handler to start dragging the thumb, and a boolean indicating if the thumb is currently being dragged.
+ */
+export function useSlider({
+  sliderRef,
+  value,
+  min,
+  max,
+  step,
+  disabled,
+  onChange,
+}: useSliderProps) {
+  const [thumbPosition, setThumbPosition] = useState(computeFirstThumbPosition(min, max, value));
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleStart = () => {
+    setIsDragging(true);
+  };
+
+  const handleEnd = () => {
+    setIsDragging(false);
+  };
+
+  const handleMove = (event: MouseEvent | TouchEvent) => {
+    event.preventDefault();
+    if (isDragging && !disabled && max > min && onChange) {
+      const roundedPercentage = computeThumbPosition(event, sliderRef, step, min, max);
+      setThumbPosition(roundedPercentage);
+
+      const newValue = computeNewValue(roundedPercentage, max, min, step);
+      onChange(newValue);
+    }
+  };
+
+  const handleClick = (event: MouseEvent) => {
+    event.preventDefault();
+    if (!disabled && max > min && onChange) {
+      const roundedPercentage = computeThumbPosition(event, sliderRef, step, min, max);
+      setThumbPosition(roundedPercentage);
+
+      const newValue = computeNewValue(roundedPercentage, max, min, step);
+      onChange(newValue);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mouseup', handleEnd);
+    document.addEventListener('mousemove', handleMove);
+    document.addEventListener('touchend', handleEnd);
+    document.addEventListener('touchmove', handleMove, { passive: false });
+    sliderRef?.current?.addEventListener('click', handleClick);
+
+    return () => {
+      document.removeEventListener('mouseup', handleEnd);
+      document.removeEventListener('mousemove', handleMove);
+      document.removeEventListener('touchend', handleEnd);
+      document.removeEventListener('touchmove', handleMove);
+      sliderRef?.current?.removeEventListener('click', handleClick);
+    };
+  }, [isDragging]);
```

Comment:
verify that placing this dependency does not make the useEffect be called too often